app/models/katello/concerns/operatingsystem_extensions.rb

```diff
@@ -44,6 +44,23 @@ def set_atomic_attributes
       def atomic?
         name.match(/.*atomic.*/i)
       end
+
+      def rhel_eos_schedule(arch_name: nil)
```

Comment:
We could have a better name for this method 😄 
It is really `rhel_release` or `rhel_release_index`. Definitely doesn't look like a schedule :)

Reply:
I originally had it named something similar, but changed it to this. It does refer to the keys of the `RHEL_EOS_SCHEDULES` elsewhere in this PR:

```rb
RHEL_EOS_SCHEDULES = { # dates that each support category ends
      'RHEL9' => {
 'full_support' => Date.parse('2027-05-31'),
        'maintenance_support' => Date.parse('2032-05-31'),
        'extended_support' => Date.parse('2035-05-31')
 ...
```

For example, here, "RHEL9" is the name of the EOS schedule with those schedule dates.

I thought "release" wasn't as accurate because we have EOS schedules for `'RHEL7'`, `'RHEL7 (System z (Structure A))'`, and `'RHEL7 (ARM)'` and those are three different architectures of the same release.

Naming it a "schedule" also opens up the possibility of adding more EOS schedules in the future that are not necessarily the same across an entire release-- say, for example, if we wanted different support schedules in different regions, we could add `'RHEL10 (North America)'` and `'RHEL10 (EMEA)'`.

Reply:
How about `rhel_eos_schedule_index`?